src/modules/esl-utils/dom/scroll/utils.ts

```diff
@@ -40,8 +40,8 @@ export type ScrollLockOptions = {
  * */
 export function lockScroll(target: Element = $html, options: ScrollLockOptions = {}): void {
   if (options.initiator) {
-    initiatorSet.add(options.initiator);
-    if (initiatorSet.size === 0) return;
+    if (initiatorMap.get(target)) return;
+    initiatorMap.set(target, options.initiator);
```

Comment:
The concurrency does not handled correctly, please check that the following test case is covered:

```ts
    describe('lockScroll handles concurrency on the single target correct, when multiple initiators are in acton', () => {
       const target = document.createElemaent('div');
       const first = document.createElemaent('div');
       const second = document.createElemaent('div');
       
       test('Target initially unlocked', () => expect(isScrollLocked(target)).toBe(false));
       
       lockScroll(target, {initiatior: first});
       test('The first initiator has locked the target', () => expect(isScrollLocked(target)).toBe(true));
       
       lockScroll(target, {initiatior: second});
       test('The second initiator has locked the target', () => expect(isScrollLocked(target)).toBe(true));
       
       unlockScroll(target, {initiatior: first});
       test('The first initiator has unlocked the target, but the target still locked by the second one', () => expect(isScrollLocked(target)).toBe(true));
       
       unlockScroll(target, {initiatior: second});
       test('The second initiator has completely unlocked the target', () => expect(isScrollLocked(target)).toBe(true));
    });
```